### PR TITLE
Extract the common logic to calculate the initial flex lines 

### PR DESCRIPTION
**Pull request by thagikura**
This PR extracts the common logic to calculate the initial flex lines from the FlexboxLayout to FlexboxHelper to re-use it from the FlexboxLayout and the FlexboxLayoutManager.

### CODE REVIEW

**Comment by thagikura** [comment]
@yaraki could you take a look?

Thanks in advance.

File: flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java [hunk 1]
```diff
@@ -159,4 +159,74 @@
...
+
+    /**
+     * @return the top padding of the flex container.
+     */
+    int getPaddingTop();
+
+    /**
+     * @return the left padding of the flex container.
+     */
+    int getPaddingLeft();
```
Comment (yaraki) [inline flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java]: Shouldn't we also add `getPaddingStart` and `getPaddingEnd`?

File: flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java [hunk 2]
```diff
@@ -159,4 +159,74 @@
...
+
+    /**
+     * Callback when a new flex item is added to the current container
+     *
+     * @param childAbsoluteIndex           the absolute index of the flex item added
+     * @param childRelativeIndexInFlexLine the relative index of the flex item added within the
+     *                                     flex line
+     * @param flexLine                     the flex line where the new flex item is added
+     */
+    void onNewFlexItemAdded(int childAbsoluteIndex, int childRelativeIndexInFlexLine,
```
Comment (yaraki) [inline flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java]: Not sure if we should differentiate these with "absolute" and "relative". Can these just be `index` and `indexInFlexLine`?

File: flexbox/src/main/java/com/google/android/flexbox/FlexItem.java [hunk 3]
```diff
@@ -238,4 +238,24 @@
...
+
+    /**
+     * @return the left margin of the flex item.
+     */
+    int getLeftMargin();
+
+    /**
+     * @return the top margin of the flex item.
+     */
+    int getTopMargin();
```
Comment (yaraki) [inline flexbox/src/main/java/com/google/android/flexbox/FlexItem.java]: Why `getPaddingTop" and `getTopMargin`?
Can this just be `getMarginTop`?

File: flexbox/src/main/java/com/google/android/flexbox/FlexItem.java [hunk 4]
```diff
@@ -238,4 +238,24 @@
      * @param flexBasisPercent the order attribute
      */
     void setFlexBasisPercent(float flexBasisPercent);
+
+    /**
+     * @return the left margin of the flex item.
+     */
+    int getLeftMargin();
```
Comment (yaraki) [inline flexbox/src/main/java/com/google/android/flexbox/FlexItem.java]: `start` and `end` variants?

**Comment by yaraki** [comment]
This is not a new topic, but there are many "new"s in the layout pass. This can cause bad performance especially for RecyclerView scrolling quickly. We might need to consider making FlexLine and other layout models behave more like `Rect` in the framework, so that we can reuse layout models without newing items each time.

File: flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java (hunk 1 above)
Comment (thagikura) [inline flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java]: Done

File: flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java (hunk 2 above)
Comment (thagikura) [inline flexbox/src/main/java/com/google/android/flexbox/FlexContainer.java]: Good point. Yes, just index and indexInFlexLine are understandable.

File: flexbox/src/main/java/com/google/android/flexbox/FlexItem.java (hunk 3 above)
Comment (thagikura) [inline flexbox/src/main/java/com/google/android/flexbox/FlexItem.java]: Renamed those to getMargin~~~~

File: flexbox/src/main/java/com/google/android/flexbox/FlexItem.java (hunk 4 above)
Comment (thagikura) [inline flexbox/src/main/java/com/google/android/flexbox/FlexItem.java]: Done.

**Comment by thagikura** [comment]
Thanks! Addressed all. PTAL @yaraki 

Regarding the many "new"s, you are right. FlexboxHelper should avoid such many "new"s I'll address it in another PR.
Thanks.

**Comment by yaraki** [comment]
+2!

**Comment by thagikura** [comment]
Thanks! Merged.

**Pull request merged**